Sources/Amplitude/Plugins/iOS/UIKitUserInteractions.swift

```diff
@@ -0,0 +1,153 @@
+#if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
+import UIKit
+
+class UIKitUserInteractions {
+    fileprivate static let amplitudeInstances = NSHashTable<Amplitude>.weakObjects()
+
+    private static let queue = DispatchQueue(label: "com.amplitude.autocapture", target: .global(qos: .utility))
+
+    private static let initializeSwizzle: () = {
+        swizzleSendAction()
+    }()
+
+    static func register(_ amplitude: Amplitude) {
+        queue.sync {
+            amplitudeInstances.add(amplitude)
+        }
+        initializeSwizzle
+    }
+
+    private static func swizzleSendAction() {
+        let applicationClass = UIApplication.self
+
+        let originalSelector = #selector(UIApplication.sendAction)
+        let swizzledSelector = #selector(UIApplication.amp_sendAction)
+
+        guard
+            let originalMethod = class_getInstanceMethod(applicationClass, originalSelector),
+            let swizzledMethod = class_getInstanceMethod(applicationClass, swizzledSelector)
+        else { return }
+
+        let originalImp = method_getImplementation(originalMethod)
+        let swizzledImp = method_getImplementation(swizzledMethod)
+
+        class_replaceMethod(applicationClass,
+            swizzledSelector,
+            originalImp,
+            method_getTypeEncoding(originalMethod))
+        class_replaceMethod(applicationClass,
+            originalSelector,
+            swizzledImp,
+            method_getTypeEncoding(swizzledMethod))
+    }
+}
+
+extension UIApplication {
+    private var keyWindow: UIWindow? {
+        connectedScenes
+            .compactMap { $0 as? UIWindowScene }
+            .flatMap { $0.windows }
+            .first { $0.isKeyWindow }
+    }
+
+    @objc dynamic func amp_sendAction(_ action: Selector, to target: Any?, from sender: Any?, for event: UIEvent?) -> Bool {
```

Comment:
why dynamic?

Reply:
My understanding is that calling the swizzled method from Objective-C and Swift could produce different results without using dynamic. Using dynamic tells Swift to always use Objective-C's dynamic dispatch.
https://stackoverflow.com/a/40796924/25576245

But, since we are swizzling the method using Swift directly and not through an Objective-C framework, it might be unnecessary to use dynamic. What are your thoughts on this?

Reply:
ah, so dynamic forces objc style dispatch even when called from Swift. But, this is being swizzled into an objc Object and should only be called from ObjC. Can you think of any cases that would break that behavior? (maybe Apple rewrites UIApplication in Swift!).  If there aren't any, probably best to remove, otherwise, please leave a comment.

Reply:
I see. One case might be directly calling the swizzled method for testing in Swift. Otherwise, I don't see any other cases here. (We don't have any tests for this scenario, so we can remove it for now.)